api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -443,6 +445,35 @@ private void createConfigFile(org.pmiops.workbench.firecloud.model.Workspace fcW
     return ResponseEntity.ok(new EmptyResponse());
   }
 
+  @Override
+  public ResponseEntity<List<FileDetail>> getNoteBookList(String workspaceNamespace,
+      String workspaceId) {
+    List<String> bucketFileList = new ArrayList<String>();
+    List<FileDetail> fileList = new ArrayList<FileDetail>();
+    try {
+      org.pmiops.workbench.firecloud.model.Workspace fireCloudWorkspace =
+          fireCloudService.getWorkspace(workspaceNamespace, workspaceId)
+              .getWorkspace();
+      String bucketName = fireCloudWorkspace.getBucketName();
+      bucketFileList = cloudStorageService.getBucketFileList(bucketName, "notebook");
+
+      if (bucketFileList != null && bucketFileList.size() > 0) {
+        bucketFileList.stream()
+            .filter(bucketFileName -> bucketFileName.matches("([^\\s]+(\\.(?i)(ipynb))$)"))
+            .forEach(bucketFileName -> {
+              FileDetail fileDetail = new FileDetail();
+              fileDetail.setName(bucketFileName);
+              fileDetail.setPath("gs://" + bucketName + "/" + bucketFileName);
+              fileList.add(fileDetail);
+            });
+      }
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      throw new NotFoundException(String.format("Workspace %s/%s not found",
```

Comment:
I would check the status code; Firecloud returns plenty of transient errors as ApiException as well and return ServerErrorException if it doesn't match whatever codes you were expecting.. (not found?)